src/box/xrow.c

```diff
@@ -2541,6 +2567,27 @@ xrow_encode_vclock_ignore0(struct xrow_header *row, const struct vclock *vclock)
 	xrow_encode_replication_request(row, &base_req, IPROTO_OK);
 }
 
+void
+xrow_encode_vclock(struct xrow_header *row, const struct vclock *vclock)
+{
+	memset(row, 0, sizeof(*row));
+	row->type = IPROTO_OK;
+	size_t size = 0;
+	size_t map_size = 1;
+	size += mp_sizeof_uint(IPROTO_VCLOCK);
+	size += mp_sizeof_vclock(vclock);
+	size += mp_sizeof_map(map_size);
+	char *buf = xregion_alloc(&fiber()->gc, size);
+	char *data = buf;
+	data = mp_encode_map(data, map_size);
+	data = mp_encode_uint(data, IPROTO_VCLOCK);
+	data = mp_encode_vclock(data, vclock);
+	assert(data <= buf + size);
+	row->body[0].iov_base = buf;
+	row->body[0].iov_len = (data - buf);
+	row->bodycnt = 1;
+}
+
 int
```

Comment:
Please, reuse `replication_request` + `xrow_encode_replication_request`. You can extend `replication_request` with ignore0 flag, for example. Or make it have two optional fields - `struct vclock *vclock;` and `struct vclock *vclock_no0;`.

Reply:
Reused the `replication_request`. Introduced `vclock_ignore0` and `vclock` in order to be consistent with function naming